packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/cap.dart

```diff
@@ -44,11 +71,34 @@ class Cap {
     double refWidth = 10,
   }) {
     assert(refWidth > 0.0);
-    return Cap._(<Object>['customCap', bitmapDescriptor.toJson(), refWidth]);
+    return CustomCap(bitmapDescriptor, refWidth);
   }
 
-  final Object _json;
+  final CapType _type;
 
   /// Converts this object to something serializable in JSON.
-  Object toJson() => _json;
+  Object toJson() => <Object>[_type.name];
+}
+
+/// CustomCap with a bitmap overlay centered at the start or
+/// end vertex of a [Polyline], orientated according to the direction of the line's
+/// first or last edge and scaled with respect to the line's stroke width.
+class CustomCap extends Cap {
+  /// [bitmapDescriptor] must not be null.
+  ///
+  /// [refWidth] is the reference stroke width (in pixels) - the stroke width for which
+  /// the cap bitmap at its native dimension is designed. Must be positive. Default value
+  /// is 10 pixels.
+  const CustomCap(this.bitmapDescriptor, [this.refWidth = 10])
```

Comment:
Given that this is a public API we will ~never be able to change, let's make it a named parameter rather than an optional positional parameter, as named parameters can much more easily be added to later.